crates/types/src/schema/metadata/mod.rs

```diff
@@ -0,0 +1,566 @@
+    #[serde(
+        with = "serde_with::As::<Option<restate_serde_util::DurationString>>",
+        skip_serializing_if = "Option::is_none",
+        default
+    )]
+    workflow_completion_retention: Option<Duration>,
+
+    /// The journal retention. When set, this applies to all requests to all handlers of this service.
+    ///
+    /// In case the request has an idempotency key, the `idempotency_retention` caps the maximum `journal_retention` time.
+    /// In case the request targets a workflow handler, the `workflow_completion_retention` caps the maximum `journal_retention` time.
+    #[serde(
+        with = "serde_with::As::<Option<restate_serde_util::DurationString>>",
+        skip_serializing_if = "Option::is_none",
+        default
+    )]
+    journal_retention: Option<Duration>,
+
+    /// This timer guards against stalled service/handler invocations. Once it expires,
+    /// Restate triggers a graceful termination by asking the service invocation to
+    /// suspend (which preserves intermediate progress).
+    ///
+    /// The 'abort timeout' is used to abort the invocation, in case it doesn't react to
+    /// the request to suspend.
+    ///
+    /// This overrides the default inactivity timeout set in invoker options.
+    #[serde(
+        with = "serde_with::As::<Option<restate_serde_util::DurationString>>",
+        skip_serializing_if = "Option::is_none",
+        default
+    )]
+    inactivity_timeout: Option<Duration>,
+
+    /// This timer guards against stalled service/handler invocations that are supposed to
+    /// terminate. The abort timeout is started after the 'inactivity timeout' has expired
+    /// and the service/handler invocation has been asked to gracefully terminate. Once the
+    /// timer expires, it will abort the service/handler invocation.
+    ///
+    /// This timer potentially **interrupts** user code. If the user code needs longer to
+    /// gracefully terminate, then this value needs to be set accordingly.
+    ///
+    /// This overrides the default abort timeout set in invoker options.
+    #[serde(
+        with = "serde_with::As::<Option<restate_serde_util::DurationString>>",
+        skip_serializing_if = "Option::is_none",
+        default
+    )]
+    abort_timeout: Option<Duration>,
+
+    /// If true, lazy state will be enabled for all invocations to this service.
+    /// This is relevant only for Workflows and Virtual Objects.
+    #[serde(skip_serializing_if = "Option::is_none", default)]
+    enable_lazy_state: Option<bool>,
+
+    /// This is a cache for the computed value of ServiceOpenAPI
+    #[serde(skip)]
+    service_openapi_cache: Arc<ArcSwapOption<ServiceOpenAPI>>,
+}
+
+impl MapAsVecItem for ServiceRevision {
+    type Key = String;
+
+    fn key(&self) -> Self::Key {
+        self.name.clone()
+    }
+}
+
+impl ServiceRevision {
+    fn as_service_metadata(&self, deployment_id: DeploymentId) -> service::ServiceMetadata {
```

Comment:
Same comment as above in regards to naming and semantics. I'm not sure when/where this conversion is being used but is it possible to move self or does this always get used in context where a copy is needed?